Efedotov/2025 03 01 terraform azure rm for adbs cross region disaster recovery (#30167)
terraform-providers/terraform-provider-azurerm: website/docs/r/oracle_autonomous_database_cross_region_disaster_recovery.html.markdown

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
description: |-
  Manages an Cross Region Disaster Recovery Autonomous Database.
---

# azurerm_oracle_autonomous_database_cross_region_disaster_recovery

Manages an Cross Region Disaster Recovery Autonomous Database.
Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with specific Cross Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with it's original database. All parameters, except "name" and "display_name" must be exactly the same as for original database or creation of Cross Region Disaster Recovery Autonomous Database will fail.

## Example Usage

```hcl

> resource "azurerm_oracle_autonomous_database_cross_region_disaster_recovery" "adbs_secondary_crdr" {

Reviewer: Would you please use a complete example here instead of using the placeholders.

  auto_scaling_enabled                = true
  auto_scaling_for_storage_enabled    = true
  mtls_connection_required            = false
  compute_model                       = "example"
  compute_count                       = 1.23456
  db_workload                         = "example"
  db_version                          = "example"
  admin_password                      = "example"
  character_set                       = "example"
  national_character_set              = "example"
}

```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Cross Region Disaster Recovery Autonomous Database. Changing this forces a new Cross Region Disaster Recovery Autonomous Database to be created.

* `display_name` - (Required) The user-friendly name for the Autonomous Database. The name does not have to be unique.

* `location` - (Required) The Azure Region where the Cross Region Disaster Recovery Autonomous Database should exist. Azure Region must be different from region where your original Autonomous Database resides. Changing this forces a new Cross Region Disaster Recovery Autonomous Database to be created.

> * `database_type` - (Required) The type of Autonomous Database. To create Cross Region Disaster Recovery Autonomous Database set this value to "CrossRegionDisasterRecovery".

Reviewer: For the properties that are marked as "ForceNew". Please add "Changing this forces a new resource to be created." to the description. 

Please also check:
name
database_type
source
remote_disaster_recovery_type
admin_password
backup_retention_period_in_days
character_set
compute_model
db_version
db_workload
display_name
mtls_connection_required
license_model
national_character_set
subnet_id
virtual_network_id
customer_contacts

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
description: |-
  Manages an Cross Region Disaster Recovery Autonomous Database.
---

# azurerm_oracle_autonomous_database_cross_region_disaster_recovery

Manages Cross Region Disaster Recovery Autonomous Database.
Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with it's original database.

> ## Example Usage

Reviewer 2: This example doesn't plan (even after removing the duplicate resource registration that is causing a panic)

```
╷
│ Error: name must contain only letters and numbers
│
│   with azurerm_oracle_autonomous_database.primary,
│   on main.tf line 35, in resource "azurerm_oracle_autonomous_database" "primary":
│   35:   name                             = "example-primary"
│
╵
╷
│ Error: display_name must contain only letters and numbers
│
│   with azurerm_oracle_autonomous_database.primary,
│   on main.tf line 36, in resource "azurerm_oracle_autonomous_database" "primary":
│   36:   display_name                     = "Example Primary"
│
```

resource "azurerm_virtual_network" "dr_vnet" {
  name                = "dr-vnet"
  location            = "westus"
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.1.0.0/16"]
}

resource "azurerm_subnet" "dr_subnet" {
  name                 = "dr-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.dr_vnet.name
  address_prefixes     = ["10.1.1.0/24"]
}

resource "azurerm_oracle_autonomous_database_cross_region_disaster_recovery" "dr_example" {
  name                          = "example-dr"
  display_name                  = "Example DR"
  location                      = "westus"
  resource_group_name           = azurerm_resource_group.example.name
  source_autonomous_database_id = azurerm_oracle_autonomous_database.primary.id
  subnet_id                     = azurerm_subnet.dr_subnet.id
  virtual_network_id            = azurerm_virtual_network.dr_vnet.id

  // Optional attributes
  replicate_automatic_backups_enabled = true

  tags = {
    Environment = "production"
    Purpose     = "disaster-recovery"
  }
}
```

## Arguments Reference

The following arguments are supported:

> * `resource_group_name` (Required) - The name of the Resource Group where the resource should exist. Changing this forces creation of a new resource.

Reviewer 2: Please use this exact phrasing for consistency with all of our other resources: "Changing this forces a new resource to be created"

```suggestion
* `resource_group_name` (Required) - The name of the Resource Group where the resource should exist. Changing this forces a new resource to be created.
```

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "eastus"
}

resource "azurerm_virtual_network" "example_primary_vnet" {
  name                = "example-primary-vnet"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.0.0.0/16"]
}

resource "azurerm_subnet" "example_primary_subnet" {
  name                 = "example-primary-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example_primary_vnet.name
  address_prefixes     = ["10.0.1.0/24"]
}

resource "azurerm_virtual_network" "example_dr_vnet" {
  name                = "example-dr-vnet"
  location            = "westus"
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.1.0.0/16"]
}

resource "azurerm_subnet" "example_dr_subnet" {
  name                 = "example-dr-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example_dr_vnet.name
  address_prefixes     = ["10.1.1.0/24"]
}

> 

Reviewer 3: The acctest has delegation settings for these subnets, while they are not declared here. Can you double check if they are missed?

Reviewer 4: delegation added to the example.

  location                      = "westus"
  display_name                  = "exampledr"
  source_autonomous_database_id = azurerm_oracle_autonomous_database.example_primary.id
  subnet_id                     = azurerm_subnet.example_dr_subnet.id

}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name for this Cross Region Disaster Recovery Autonomous Database. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the resource should exist. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the Cross Region Disaster Recovery Autonomous Database should exist. Changing this forces a new resource to be created.

* `display_name` - (Required) The user-friendly name for the Autonomous Database. Changing this forces a new resource to be created.

* `source_autonomous_database_id` - (Required) The ID of the source (primary) Autonomous Database. Changing this forces a new resource to be created.

* `subnet_id` - (Required) The ID of the subnet in the target region. Changing this forces a new resource to be created.

> * `replicate_automatic_backups_enabled` - (Optional) If true, 7 days of backups are replicated across regions. Default value `false`. Changing this forces a new resource to be created.

Reviewer 3: ```suggestion
* `replicate_automatic_backups_enabled` - (Optional) If true, 7 days of backups are replicated across regions. Defaults to `false`. Changing this forces a new resource to be created.
```

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
description: |-
  Manages an Cross Region Disaster Recovery Autonomous Database.
---

# azurerm_oracle_autonomous_database_cross_region_disaster_recovery

Manages Cross Region Disaster Recovery Autonomous Database.

> Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with it's original database. All parameters, except "name" and "display_name" must be exactly the same as for original database or creation of Cross Region Disaster Recovery Autonomous Database will fail.

Reviewer 4: @sreallymatt this is not the case for backup and clone databases.

> Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with it's original database. All parameters, except "name" and "display_name" must be exactly the same as for original database or creation of Cross Region Disaster Recovery Autonomous Database will fail.

Reviewer 4: "All parameters, except "name" and "display_name" must be exactly the same" and "Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database" also contradict each other as location should be provided with a different value

removed this statement as its handled by design.

* `display_name` - (Required) The user-friendly name for the Autonomous Database. The name does not have to be unique. Changing this forces is a new resource to be created.

* `location` - (Required) The Azure Region where the Cross Region Disaster Recovery Autonomous Database should exist. Azure Region must be different from a region where your original Autonomous Database resides. Changing this forces a new Cross Region Disaster Recovery Autonomous Database to be created.

* `database_type` - (Required) The type of Autonomous Database. To create Cross Region Disaster Recovery Autonomous Database set this value to "CrossRegionDisasterRecovery." Changing this forces a new Cross-Region Disaster Recovery Autonomous Database to be created.

* `source` - (Required) Source of Autonomous Database. To create Cross Region Disaster Recovery Autonomous Database set this value to "CrossRegionDisasterRecovery." Changing this forces a new Cross-Region Disaster Recovery Autonomous Database to be created.

> * `remote_disaster_recovery_type` - (Required) Type of recovery. Value can be either `Adg` (Autonomous Data Guard) or `BackupBased`.Changing this forces a new Cross Region Disaster Recovery Autonomous Database to be created.

Reviewer 4: removed

  auto_scaling_enabled             = false
  auto_scaling_for_storage_enabled = false
  backup_retention_period_in_days  = 7
  character_set                    = "AL32UTF8"
  compute_count                    = 2
  compute_model                    = "ECPU"
  customer_contacts                = ["test@example.com"]
  data_storage_size_in_tbs         = 1
  db_version                       = "19c"
  db_workload                      = "DW"
  license_model                    = "LicenseIncluded"
  mtls_connection_required         = false
  national_character_set           = "AL16UTF16"
}

resource "azurerm_oracle_autonomous_database_cross_region_disaster_recovery" "example" {
  name                                = "exampledr"
  resource_group_name                 = azurerm_resource_group.example.name
  location                            = "westus"
  display_name                        = "exampledr"
  source_autonomous_database_id       = azurerm_oracle_autonomous_database.example_primary.id
  subnet_id                           = azurerm_subnet.example_dr_subnet.id
  replicate_automatic_backups_enabled = true

  tags = {
    environment = "production"

>   }

Reviewer 5: typically we only include Required properties in the example, so `replicate_automatic_backups_enabled` and `tags` can be omitted

Reviewer 4: @ms-zhenhua our customers needs to see everything in the example thats why it was added, or do you think we should add another example with all the supported fields and keep this one to the required ones?

Reviewer 3: The default example here meant to be as minimal as possible. If you want to demonstrate some other combinations of different properties, you are free to add additional examples below the default one.

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
description: |-
  Manages a Cross Region Disaster Recovery Autonomous Database.
---

# azurerm_oracle_autonomous_database_cross_region_disaster_recovery

Manages a Cross Region Disaster Recovery Autonomous Database.
A Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of the Autonomous Database for which you want to create a Disaster Recovery instance. The Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from the region of the main Autonomous Database. You must create a separate virtual network and subnet in this second region for the Cross Region Disaster Recovery Autonomous Database to be able to communicate with its original database.

## Example Usage

```hcl

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "eastus"
}

resource "azurerm_virtual_network" "example_vnet" {

>   name                = "exampleVnet"

Reviewer 5: ```suggestion
  name                = "example-vnet"
```

  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.0.0.0/16"]
}

resource "azurerm_subnet" "example_subnet" {
  name                 = "exampleSubnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example_vnet.name
  address_prefixes     = ["10.0.1.0/24"]
}

resource "azurerm_oracle_autonomous_database" "primary" {
  name                             = "examplePrimary"
  display_name                     = "ExamplePrimary"
  resource_group_name              = azurerm_resource_group.example.name
  location                         = "eastus"
  compute_model                    = "ECPU"
  compute_count                    = 2
  license_model                    = "LicenseIncluded"
  backup_retention_period_in_days  = 7
  auto_scaling_enabled             = true
  auto_scaling_for_storage_enabled = true
  mtls_connection_required         = false

>   data_storage_size_in_tbs         = 1

Reviewer 4: same like the above.

  db_workload                      = "DW"
  admin_password                   = "SomeP@ssw0rd123"
  db_version                       = "19c"
  character_set                    = "AL32UTF8"
  national_character_set           = "AL16UTF16"
  subnet_id                        = azurerm_subnet.example_subnet.id
  virtual_network_id               = azurerm_virtual_network.example_vnet.id
  customer_contacts                = ["test@example.com"]
}

resource "azurerm_virtual_network" "dr_vnet" {

>   name                = "exampleDrVnet"

Reviewer 5: ```suggestion
  name                = "example-dr-vnet"
```

  location            = "westus"
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.1.0.0/16"]
}

resource "azurerm_subnet" "dr_subnet" {

>   name                 = "exampleDrSubnet"

Reviewer 5: ```suggestion
  name                 = "example-dr-subnet"
```

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_oracle_autonomous_database_cross_region_disaster_recovery"
description: |-
  Manages an Cross Region Disaster Recovery Autonomous Database.
---

# azurerm_oracle_autonomous_database_cross_region_disaster_recovery

> Manages Cross Region Disaster Recovery Autonomous Database.

Reviewer 5: ```suggestion
Manages a Cross Region Disaster Recovery Autonomous Database.
```

> Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with it's original database.

Reviewer 5: ```suggestion
Cross Region Disaster Recovery Autonomous Database is an Autonomous Database with a specific Cross-Region Disaster Recovery role. It must be an exact copy of Autonomous Database for which you want to create a Disaster Recovery instance. Cross Region Disaster Recovery Autonomous Database must reside in a region that is different from region of main Autonomous Database. You must create a separate virtual network and subnet in this second region for Cross Region Disaster Recovery Autonomous Database to be able to communicate with its original database.
```

  location                      = "westus"
  resource_group_name           = azurerm_resource_group.example.name
  source_autonomous_database_id = azurerm_oracle_autonomous_database.primary.id
  subnet_id                     = azurerm_subnet.dr_subnet.id
  virtual_network_id            = azurerm_virtual_network.dr_vnet.id

  // Optional attributes
  replicate_automatic_backups_enabled = true

  tags = {
    Environment = "production"
    Purpose     = "disaster-recovery"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `resource_group_name` - (Required) The name of the Resource Group where the resource should exist. Changing this forces a new resource to be created.

* `name` - (Required) The name for this Cross Region Disaster Recovery Autonomous Database. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the Cross Region Disaster Recovery Autonomous Database should exist. Must differ from the primary region. Changing this forces a new resource to be created.

* `display_name` - (Required) The user-friendly name for the Autonomous Database. Changing this forces a new resource to be created.

> * `display_name` - (Required) The user-friendly name for the Autonomous Database. Changing this forces a new resource to be created.

Reviewer 5: ```suggestion
* `display_name` - (Required) The user-friendly name for the Autonomous Database. Changing this forces a new resource to be created.
```

> * `source_autonomous_database_id` - (Required) The Azure Resource ID of the source (primary) Autonomous Database. Changing this forces a new resource to be created.

Reviewer 5: ```suggestion
* `source_autonomous_database_id` - (Required) The ID of the source (primary) Autonomous Database. Changing this forces a new resource to be created.
```

> * `subnet_id` - (Required) The Azure Resource ID of the subnet in the target region. Changing this forces a new resource to be created.

Reviewer 5: ```suggestion
* `subnet_id` - (Required) The ID of the subnet in the target region. Changing this forces a new resource to be created.
```
